src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -163,6 +163,11 @@ public function load(array $configs, ContainerBuilder $container)
             $definition->replaceArgument(1, null);
         }
 
+        $this->addAnnotatedClassesToCache(array(
+            'Bundle\\Controller\\',
+            'Bundle\\Entity\\',
```

Comment:
Much more classes can be annotated (and you cannot hard-code a list). Even worse, much more classes can be *checked for annotations* (even though they don't use them).
Your change means that all these classes will miss the cache forever (as the cache is read-only), which makes things much worse actually.